provisioning/baremetal_pod.go

```diff
@@ -0,0 +1,353 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"k8s.io/utils/pointer"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+)
+
+const (
+	baremetalSharedVolume      = "metal3-shared"
+	metal3AuthRootDir          = "/auth"
+	ironicCredentialsVolume    = "metal3-ironic-basic-auth"
+	inspectorCredentialsVolume = "metal3-inspector-basic-auth"
+	htpasswdEnvVar             = "HTTP_BASIC_HTPASSWD" // #nosec
+)
+
+var sharedVolumeMount = corev1.VolumeMount{
+	Name:      baremetalSharedVolume,
+	MountPath: "/shared",
+}
+
+var ironicCredentialsMount = corev1.VolumeMount{
+	Name:      ironicCredentialsVolume,
+	MountPath: metal3AuthRootDir + "/ironic",
+	ReadOnly:  true,
+}
+
+var inspectorCredentialsMount = corev1.VolumeMount{
+	Name:      inspectorCredentialsVolume,
+	MountPath: metal3AuthRootDir + "/ironic-inspector",
+	ReadOnly:  true,
+}
+
+func buildEnvVar(name string, baremetalProvisioningConfig *metal3iov1alpha1.ProvisioningSpec) corev1.EnvVar {
+	value := getMetal3DeploymentConfig(name, baremetalProvisioningConfig)
+	if value != nil {
+		return corev1.EnvVar{
+			Name:  name,
+			Value: *value,
+		}
+	} else {
+		return corev1.EnvVar{
+			Name: name,
+		}
+	}
+}
+
+func setMariadbPassword() corev1.EnvVar {
```

Comment:
Could these environment variable definitions be constants, like the volume mounts are?

Reply:
Done

Reply:
What I meant was, move lines 66-75 up to the top of the file (say around line 50) and define them as variables like `inspectorCredentialsMount` instead of using a function that returns a new object created from constant data each time.

Reply:
Done. Added vars for marladbPassword and metal3Volumes instead of methods.